### PR TITLE
Fix installation on guides/source/action_text_overview.md

### PR DESCRIPTION
**Pull request by shioimm**
[Active Storage Overview](https://edgeguides.rubyonrails.org/active_storage_overview.html#transforming-images)
gem 'image_proccessing' is required.

### Summary
I used ActionText with reference to [Action Text Overview](https://edgeguides.rubyonrails.org/action_text_overview.html).
I received the following error when I attached an image by drag and drop.

```
Started GET "/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--15600b0b846815065144f9d5898134e62ac79eaf/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9VY21WemFYcGxYM1J2WDJ4cGJXbDBXd2RwQWdBRWFRSUFBdz09IiwiZXhwIjpudWxsLCJwdXIiOiJ2YXJpYXRpb24ifX0=--bdd56a50d888ef5d9416578248b1b579125cfce1/Screen%20Shot%202019-04-26%20at%2020.04.50.png" for ::1 at 2019-04-27 14:41:50 +0900
Processing by ActiveStorage::RepresentationsController#show as PNG
  Parameters: {"signed_blob_id"=>"eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBCZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--15600b0b846815065144f9d5898134e62ac79eaf", "variation_key"=>"eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9VY21WemFYcGxYM1J2WDJ4cGJXbDBXd2RwQWdBRWFRSUFBdz09IiwiZXhwIjpudWxsLCJwdXIiOiJ2YXJpYXRpb24ifX0=--bdd56a50d888ef5d9416578248b1b579125cfce1", "filename"=>"Screen Shot 2019-04-26 at 20.04.50"}
  ActiveStorage::Blob Load (0.2ms)  SELECT "active_storage_blobs".* FROM "active_storage_blobs" WHERE "active_storage_blobs"."id" = ? LIMIT ?  [["id", 1], ["LIMIT", 1]]
  Disk Storage (0.2ms) Checked if file exists at key: variants/irz0bykwoiwpg2rq94ks6e45hjpz/a2845b7f0f3bf59332871c37aa09358e5c3516a2ad65fc279472c39200d7e664 (no)
  Disk Storage (1.4ms) Downloaded file from key: irz0bykwoiwpg2rq94ks6e45hjpz
DEPRECATION WARNING: Generating image variants will require the image_processing gem in Rails 6.1. Please add `gem 'image_processing', '~> 1.2'` to your Gemfile. (called from transformer at /Users/misakishioi/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/activestorage-6.0.0.rc1/app/models/active_storage/variation.rb:64)
Completed 500 Internal Server Error in 26ms (ActiveRecord: 0.2ms | Allocations: 6044)

LoadError (cannot load such file -- mini_magick):
```
I wasn't aware that gem 'image_processing' was required until this error appeared.

---

Referring to this https://github.com/rails/rails/pull/36117#discussion_r291427669  and https://github.com/rails/rails/pull/36117#discussion_r291798436, I added the link to [Action Storage Overview](https://edgeguides.rubyonrails.org/active_storage_overview.html#transforming-images) in Action Text Overview.

### CODE REVIEW

**Comment by prathamesh-sonpatki** [comment]
It actually hits the mini_magic_transformer. Perhaps we can do something like this while trying to require `mini_magic` and `image_processing` gems:

```rb
begin
  require "mini_magick"
rescue LoadError => e
  $stderr.puts "You don't have mini_magic installed in your application. Please add it to your Gemfile and run bundle install."
  raise e
end
```

cc @georgeclaghorn

**Comment by shioimm** [comment]
@prathamesh-sonpatki 
  cc @georgeclaghorn 

Thank you for your comment!
I noticed that image_processing relies on mini_magick.

And if you install only mini_magick without installing imege_processing, the following error occurs on mini_magick.

```
Started GET "/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--0f05a8d929333df2e88d35023c031cdba15a291b/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9VY21WemFYcGxYM1J2WDJ4cGJXbDBXd2RwQWdBRWFRSUFBdz09IiwiZXhwIjpudWxsLCJwdXIiOiJ2YXJpYXRpb24ifX0=--bdd56a50d888ef5d9416578248b1b579125cfce1/tabouhiyoko.png" for ::1 at 2019-04-27 23:13:44 +0900
Processing by ActiveStorage::RepresentationsController#show as PNG
  Parameters: {"signed_blob_id"=>"eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBDZz09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--0f05a8d929333df2e88d35023c031cdba15a291b", "variation_key"=>"eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9VY21WemFYcGxYM1J2WDJ4cGJXbDBXd2RwQWdBRWFRSUFBdz09IiwiZXhwIjpudWxsLCJwdXIiOiJ2YXJpYXRpb24ifX0=--bdd56a50d888ef5d9416578248b1b579125cfce1", "filename"=>"tabouhiyoko"}
  ActiveStorage::Blob Load (0.4ms)  SELECT "active_storage_blobs".* FROM "active_storage_blobs" WHERE "active_storage_blobs"."id" = ? LIMIT ?  [["id", 5], ["LIMIT", 1]]
  Disk Storage (0.1ms) Checked if file exists at key: variants/48jq1kqjq7koqcod7ngw2igr2bj2/a2845b7f0f3bf59332871c37aa09358e5c3516a2ad65fc279472c39200d7e664 (no)
  Disk Storage (24.0ms) Downloaded file from key: 48jq1kqjq7koqcod7ngw2igr2bj2
DEPRECATION WARNING: Generating image variants will require the image_processing gem in Rails 6.1. Please add `gem 'image_processing', '~> 1.2'` to your Gemfile. (called from transformer at /Users/misakishioi/.rbenv/versions/2.6.2/lib/ruby/gems/2.6.0/gems/activestorage-6.0.0.rc1/app/models/active_storage/variation.rb:64)
Completed 500 Internal Server Error in 187ms (ActiveRecord: 0.4ms | Allocations: 5213)
  
MiniMagick::Error (`mogrify -resize-to-limit [1024, 768] /var/folders/wz/85kpnlyx0hdbmgf4_445j9nc0000gn/T/ActiveStorage-5-20190427-18225-f3bpov.png` failed with error:
mogrify: unrecognized option `-resize-to-limit' @ error/mogrify.c/MogrifyImageCommand/5936.
):
```
From these things, I can see that image_processing must be installed in order to use ActionText, but I don't think I need to install additional mini_magick.

Could you tell me if there is anything else I missed?

**Comment by okuramasafumi** [comment]
https://edgeguides.rubyonrails.org/active_storage_overview.html#transforming-images
I can see the instruction to install `image_processing` in Active Storage Guide. Maybe it's worth adding the same instruction to Action Text Guide because we cannot expect people who like to use Action Text to refer to Active Storage Guide.

File: guides/source/action_text_overview.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ happens after every keystroke, and avoids the need to use execCommand at all.
 ## Installation
 
 Run `rails action_text:install` to add the Yarn package and copy over the necessary migration.
+Uncomment # gem 'image_processing', '~> 1.2' and run `bundle install`.
```
Comment (y-yagi) [inline guides/source/action_text_overview.md]: Since Action Text uses Active Stroage to manage attachments, users also need to set up Active Storage to use Action Text (the migration file generated but users also need to be aware of config). 
I think we need an explanation on that point. The Active Storage setup is described in the Active Storage guide, so I think it's enough to have a link to that page here. 

Can you add a description of the Active Storage setup required and a link to that page?

Comment (shioimm) [reply guides/source/action_text_overview.md]: I changed the text to add link to  Active Storage Overview.
Thank you for your suggestion!

File: guides/source/action_text_overview.md [hunk 2]
```diff
@@ -46,6 +46,8 @@ happens after every keystroke, and avoids the need to use execCommand at all.
 ## Installation
 
 Run `rails action_text:install` to add the Yarn package and copy over the necessary migration.
+Also you need to set up Active Storage to use Action Text to manage attachments.
+Please refer to the [Active Storage Overview](https://edgeguides.rubyonrails.org/active_storage_overview.html#transforming-images).
```
Comment (y-yagi) [inline guides/source/action_text_overview.md]: Users need to know about the whole Active Storage only not transform images. 
I think that the link destination is the top of Active Storage guide is more appropriate.   

Also, you don't need to specify a domain in link during guides. Only `active_storage_overview.html` is enough.

Comment (shioimm) [reply guides/source/action_text_overview.md]: Sorry I should have referred to other guide to add link.
Thank you again to suggest.

File: guides/source/action_text_overview.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ happens after every keystroke, and avoids the need to use execCommand at all.
 ## Installation
 
 Run `rails action_text:install` to add the Yarn package and copy over the necessary migration.
+Also you need to set up Active Storage to use Action Text to manage attachments.
```
Comment (y-yagi) [inline guides/source/action_text_overview.md]: I noticed that we used the `embedded images and other attachments` elsewhere in the guide. So how about? 
`Also, you need to set up Active Storage for embedded images and other attachments.`

Also, it seems that an unrelated file(`.5_2_release_notes.md.swp`) have been included. Please remove ;)

Comment (shioimm) [reply guides/source/action_text_overview.md]: I agree with you that is to the point to use `embedded images and other attachments` instead of `manage attachments`.
I fixed it and removed unnecessary swap file.　Thank you for pointing it out!

**Comment by y-yagi** [comment]
@shioimm Thanks!!

**Comment by shioimm** [comment]
@y-yagi 
Thank you very much again and again for guiding me towards the right action!
